app/pastes/create-paste/createPasteInteractor.js

```diff
@@ -0,0 +1,27 @@
+const ValidationError = require('../../errors/validationError');
+
+class CreatePasteInteractor {
+    constructor ({presenter, validator, pasteRepository, pasteFactory, idGenerator}) {
+        this.presenter = presenter;
+        this.validator = validator;
+        this.pasteRepository = pasteRepository;
+        this.pasteFactory = pasteFactory;
+        this.idGenerator = idGenerator;
+    }
+
+    async execute({name, text, visibility, expiresAfter, authorId}) {
```

Comment:
Use created request here)
Tip: never decompose class instances, even JS allows us it